oat-sa/tao-item-runner-qti-fe
# Feature/ms 147/enhanced readonly mode

Ticket: https://oat-sa.atlassian.net/browse/MS-147

### What has changed?

* A new renderer(reviewRenderer) is added with interactions and choices in readonly/disabled mode
* A new helper function is added to map IMS view property to the right renderer
* All the choices and interactions which are not needed to be changed are inherited from qtiCommonRenderer
* Input fields and text areas are in `readonly` mode
* Select option fields, checkboxes and radio buttons are `disabled`
* All the click/tap and drag/drop event handlers are disabled with `pointer-events: none` style on templates

#### Related PRs:

- https://github.com/oat-sa/extension-tao-testqti-previewer/pull/63
- https://github.com/oat-sa/extension-tao-itemqti/pull/1439
- https://github.com/oat-sa/extension-lti-outcomeui/pull/13

### How to test?

- Checkout this branch in test previewer: `feature/MS-147/enhanced-readonly-mode`
- Checkout this branch in tao-item-qti: `feature/MS-147/enhanced-readonly-mode`
- Install node modules in tao-item-qti
- Create an item with any interactions. (I have an item with all interactions ready, you can ask me for that)
- Preview the item, you should see it as expected and MUST be able to interact with all interactions
- Take the test, you should see all items as usual and MUST be able to interact with all of them.
- Now, view the result of the test you submitted, now you are viewing the test with `reviewRenderer` so you won't be able to interact with any of the interactions as it is in read-only mode.


### Checklist:

- [ ] Merge this PR and publish package
- [ ] Change dependency version here and merge this PR: https://github.com/oat-sa/extension-tao-itemqti/pull/1439
- [ ] Merge this PR: https://github.com/oat-sa/extension-tao-testqti-previewer/pull/63
- [ ] Merge this PR: https://github.com/oat-sa/extension-lti-outcomeui/pull/13
